fs/fs.go

```diff
@@ -141,10 +141,12 @@ func NewFilesystem(ctx context.Context, root string, cfg config.FSConfig, opts .
 			return docker.ConfigureDefaultRegistries(docker.WithPlainHTTP(docker.MatchLocalhost))(refspec.Hostname())
 		})
 	}
-	ctx, store, err := store.NewContentStore(ctx, store.WithType(store.ContentStoreType(cfg.ContentStoreConfig.Type)), store.WithNamespace(cfg.ContentStoreConfig.Namespace))
+	ctx, store, done, err := store.NewContentStore(ctx, store.WithType(store.ContentStoreType(cfg.ContentStoreConfig.Type)), store.WithNamespace(cfg.ContentStoreConfig.Namespace))
 	if err != nil {
 		return nil, fmt.Errorf("cannot create content store: %w", err)
 	}
+	// FIXME: This runs far too soon. Writing index.json for the soci content store should happen after each change to the store and/or when the snapshotter shuts down.
+	defer done(ctx)
```

Comment:
All the other cases are single CLI commands which perform some operation then clean up after themselves. Here we are creating the filesystem for the snapshotter which will be used for an indefinite period. This `done` should happen when the filesystem is closed, probably when the snapshotter is exiting. An equivalent operation (which wouldn't really be "done") may also be needed after each new content creation. Either or both of these options would require putting the function into the filesystem struct (or somewhere else persistent) to be called later.

I am seeking input on the desirable path forward here.

Reply:
This operation could be moved into `Push` but would then happen far more often than is reasonable for sequential operations.